src/hooks/useSession.ts

```diff
@@ -1,3 +1,4 @@
+'use server'
```

Comment:
it was complaining about how supabase was handling cookies. making it explicitly a server-side file fixed it.